### PR TITLE
Implement the Participation & Presentation service

### PR DESCRIPTION
**Pull request by Tenyitamas**

### CODE REVIEW

File: src/consultations/participation.service.ts [hunk 1]
```diff
@@ -1,7 +1,56 @@
...
+      },
+      include: {
+        consultation: true,
+      },
+    })
+
+    return participations.map(({ consultation, ...p }) => ({
+      ...p,
+      consultation,
+    }))
```
Comment (Tschonti) [inline src/consultations/participation.service.ts:25]: this is equivalent to `return participations`

File: src/consultations/participation.service.ts [hunk 2]
```diff
@@ -1,7 +1,56 @@
...
+  }
+
+  async findAllByConsultationId(consultationId: number) {
+    const participations = await this.prisma.participation.findMany({
+      where: {
+        consultationId,
+      },
+    })
```
Comment (Tschonti) [inline src/consultations/participation.service.ts:33]: you could include the user here

File: src/consultations/presentation.service.ts [hunk 3]
```diff
@@ -1,7 +1,49 @@
...
   constructor(private prisma: PrismaService) {}
+  create(data: Prisma.PresentationUncheckedCreateInput) {
+    return this.prisma.presentation.create({ data })
+  }
+  async findAllByUserId(userId: number) {
+    const presentations = await this.prisma.presentation.findMany({
+      where: {
+        userId,
+      },
+    })
```
Comment (Tschonti) [inline src/consultations/presentation.service.ts:16]: you could include consultation here

File: src/consultations/presentation.service.ts [hunk 4]
```diff
@@ -1,7 +1,49 @@
...
+
+    return presentations
+  }
+
+  async findAllByConsultationId(consultationId: number) {
+    const presentations = await this.prisma.presentation.findMany({
+      where: {
+        consultationId,
+      },
+    })
```
Comment (Tschonti) [inline src/consultations/presentation.service.ts:26]: you could include the user here

**Review by Tschonti — CHANGES_REQUESTED** [review]
Good job! Maybe we could include a few more fields to help us later on.

Also, if you don't transform the value returned by prisma, you could omit the whole async/await syntax and just do `return this.prisma.presentations.findMany(...)`

**Review by Tschonti — APPROVED** [review]

**Review by kisbogdan-kolos — APPROVED** [review]

**Review by Frigyes309 — APPROVED** [review]

**Pull request merged**